static/js/ducks/sources.js

```diff
@@ -5,87 +5,133 @@ export const FETCH_SOURCES = "skyportal/FETCH_SOURCES";
 export const FETCH_SOURCES_OK = "skyportal/FETCH_SOURCES_OK";
 export const FETCH_SOURCES_FAIL = "skyportal/FETCH_SOURCES_FAIL";
 
-export const FETCH_GROUP_SOURCES = "skyportal/FETCH_GROUP_SOURCES";
-export const FETCH_GROUP_SOURCES_OK = "skyportal/FETCH_GROUP_SOURCES_OK";
-export const FETCH_GROUP_SOURCES_FAIL = "skyportal/FETCH_GROUP_SOURCES_FAIL";
+export const FETCH_SAVED_GROUP_SOURCES = "skyportal/FETCH_SAVED_GROUP_SOURCES";
+export const FETCH_SAVED_GROUP_SOURCES_OK =
+  "skyportal/FETCH_SAVED_GROUP_SOURCES_OK";
+export const FETCH_SAVED_GROUP_SOURCES_FAIL =
+  "skyportal/FETCH_SAVED_GROUP_SOURCES_FAIL";
 
-export function fetchSources(filterParams = {}) {
+export const FETCH_PENDING_GROUP_SOURCES =
+  "skyportal/FETCH_PENDING_GROUP_SOURCES";
+export const FETCH_PENDING_GROUP_SOURCES_OK =
+  "skyportal/FETCH_PENDING_GROUP_SOURCES_OK";
+export const FETCH_PENDING_GROUP_SOURCES_FAIL =
+  "skyportal/FETCH_PENDING_GROUP_SOURCES_FAIL";
+
+const addFilterParamDefaults = (filterParams) => {
   if (!Object.keys(filterParams).includes("pageNumber")) {
     filterParams.pageNumber = 1;
   }
+  if (!Object.keys(filterParams).includes("numPerPage")) {
+    filterParams.numPerPage = 10;
+  }
+};
+
+export function fetchSources(filterParams = {}) {
+  addFilterParamDefaults(filterParams);
   const params = new URLSearchParams(filterParams);
   const queryString = params.toString();
   return API.GET(`/api/sources?${queryString}`, FETCH_SOURCES);
 }
 
-export function fetchGroupSources(filterParams = {}) {
+export function fetchSavedGroupSources(filterParams = {}) {
+  addFilterParamDefaults(filterParams);
   const params = new URLSearchParams(filterParams);
   const queryString = params.toString();
-  return API.GET(`/api/sources?${queryString}`, FETCH_GROUP_SOURCES);
+  return API.GET(`/api/sources?${queryString}`, FETCH_SAVED_GROUP_SOURCES);
+}
+
+export function fetchPendingGroupSources(filterParams = {}) {
+  addFilterParamDefaults(filterParams);
+  filterParams.pendingOnly = true;
+  const params = new URLSearchParams(filterParams);
+  const queryString = params.toString();
+  return API.GET(`/api/sources?${queryString}`, FETCH_PENDING_GROUP_SOURCES);
 }
 
 const initialState = {
-  latest: null,
-  groupSources: null,
+  sources: null,
   pageNumber: 1,
-  lastPage: false,
   totalMatches: 0,
-  numberingStart: 0,
-  numberingEnd: 0,
+  numPerPage: 10,
 };
 
-const reducer = (state = initialState, action) => {
+const reducer = (
+  state = {
+    latest: initialState,
+    savedGroupSources: initialState,
+    pendingGroupSources: initialState,
+  },
+  action
+) => {
   switch (action.type) {
     case FETCH_SOURCES: {
       return {
         ...state,
-        queryInProgress: action.parameters.body.pageNumber === undefined,
+        latest: {
+          ...state.latest,
+          queryInProgress: action.parameters.body.pageNumber === undefined,
```

Comment:
Do we need the `queryInProgress` state? Doesn't seem to be used anywhere for sources or group sources

Reply:
You're right it's not needed for the groupSources branches, but is still needed in the "latest" (non-group-specific) state for the SourceList logic. I'll remove them from the group sources reducers

Reply:
Done